src/drivers/serial/imx_uart/imx_uart.c

```diff
@@ -69,49 +153,49 @@ static int imxuart_getc(struct uart *dev) {
 }
 
 static int imxuart_putc(struct uart *dev, int ch) {
-	while (!(UART(USR2) & USR2_TXFE)) {
-	}
-
 	UART(TXR) = ch;
 
+	while (!(UART(UTS) & (1 << 6))) {
```

Comment:
Fixed constant in the following commit.
Waiting is just to make sure symbol was sent, nothing special.